workers/data_refinery_workers/downloaders/geo.py

```diff
@@ -0,0 +1,374 @@
+from __future__ import absolute_import, unicode_literals
+import urllib.request
+import os
+import shutil
+import gzip
+import tarfile
+from typing import List
+from contextlib import closing
+
+from data_refinery_common.models import (
+    DownloaderJob,
+    Experiment, 
+    Sample, 
+    ExperimentAnnotation, 
+    ExperimentSampleAssociation, 
+    OriginalFile, 
+    DownloaderJobOriginalFileAssociation,
+    OriginalFileSampleAssociation
+)
+from data_refinery_workers.downloaders import utils
+from data_refinery_common.logging import get_and_configure_logger
+from data_refinery_common.utils import get_env_variable
+
+
+logger = get_and_configure_logger(__name__)
+LOCAL_ROOT_DIR = get_env_variable("LOCAL_ROOT_DIR", "/home/user/data_store")
+
+# chunk_size is in bytes
+CHUNK_SIZE = 1024 * 256
+
+def get_miniml_url(self, experiment_accession_code):
+    """ Create the url for the MINiML file for an accession code"""
+    geo = experiment_accession_code.upper()
+    geotype = geo[:3]
+    range_subdir = sub(r"\d{1,3}$", "nnn", geo)
+
+    miniml_url_template = ("ftp://ftp.ncbi.nlm.nih.gov/geo/"
+              "{root}/{range_subdir}/{record}/miniml/{record_file}")
+    miniml_url = miniml_url_template.format(root="series",
+                        range_subdir=range_subdir,
+                        record=geo,
+                        record_file="%s_family.xml.tgz" % geo)
+
+    return raw_url
+
+def _download_file(download_url: str, file_path: str, job: DownloaderJob) -> None:
+    """ Download a file from ArrayExpress via FTP. There is no Aspera endpoint
```

Comment:
This will ultimately become simply a dispatcher function when we figure out GEO's public-if-you-ask-for-it-and-definately-not-documented Aspera endpoint, which requires a different configuration than the other Aspera downloader. Then, this function will dispatch to an FTP or Aspera downloader, but I think in both cases the configurations for optimal connections will be specific to the servers on the other end (Aspera is highly configurable, and if we do optimal FTP with [https://github.com/aria2/aria2](aria2), that will require server-specific configuration as well).

Reply:
Ok, that seems to make sense. Can you put something to that effect in the docstring?